spec/registry.xml

```diff
@@ -0,0 +1,250 @@
+    <matchSignature>
+      <option name="type" values="cardinal ordinal" default="cardinal"/>
+      <!--
+        Offset is neither ECMAScript Intl.PluralRules, nor ICU com.ibm.icu.text.PluralRules.
+        But it is implemented on top that base functionality.
+        Similar to exact values in number formatting, which does not come from NumberFormat(ter).
+        It is supported by both ICU and Fluent, and it is used for constructs like this:
+        "You invited A, B, and 2 more guests."
+      -->
+      <option name="offset" values="positiveInteger"/>
+      <!-- The minimum number of integer digits to use.
+        A value with a smaller number of integer digits than this number will be
+        left-padded with zeros (to the specified length) when formatted.
+      -->
+      <option name="minimumIntegerDigits" values="1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21" default="1"/>
+      <!-- The minimum number of fraction digits to use.
+        The default for plain number and percent formatting is 0;
+        the default for currency formatting is the number of minor unit digits provided by
+        the ISO 4217 currency code list (2 if the list doesn't provide that information).
+      -->
+      <option name="minimumFractionDigits" values="0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20"/>
+      <!-- The maximum number of fraction digits to use.
+        The default for plain number formatting is the larger of minimumFractionDigits and 3;
+        the default for currency formatting is the larger of minimumFractionDigits and the number of minor
+        unit digits provided by the ISO 4217 currency code list (2 if the list doesn't provide that information);
+        the default for percent formatting is the larger of minimumFractionDigits and 0.
+      -->
+      <option name="maximumFractionDigits" values="0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20"/>
+      <!-- The minimum number of significant digits to use. -->
+      <option name="minimumSignificantDigits" values="1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21" default="1"/>
+      <!-- The maximum number of significant digits to use. -->
+      <option name="maximumSignificantDigits" values="1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21" default="21"/>
+
+      <match pattern="anyNumber"/>
+      <match values="zero one two few many"/>
+
+    </matchSignature>
+  </function>
+
+  <function name="select">
+    <description>Literal selection</description>
+    <matchSignature>
+      <match pattern="anythingNotEmpty"/>
+    </matchSignature>
+  </function>
+
+  <!-- ICU does not have this, and one might argue it is just syntactic sugar on top of "select".
+    ICU rejected this a few times.
+  -->
+  <function name="gender">
+    <description>Gender selection</description>
+    <matchSignature>
+      <match values="feminine masculine"/>
+    </matchSignature>
+  </function>
+
```

Comment:
I don't think these should be included here. Happy to discuss them separately.

```suggestion
```

Reply:
Literal selection is really needed. Exists in ICU, exists in Fluent.
And I've seen it used a lot, very handy to have.

Removed gender, added it to issue https://github.com/unicode-org/message-format-wg/issues/422

Reply:
I would like for us to have a separate discussion about `<function name="select">`. I think we agree on the ultimate behaviour we want to enable, but disagree about some of the details in how to achieve it.

Reply:
@eemeli Create an issue and Agenda+ it.

Note that `SelectFormat` exists in ICU4J's message format and so would be part of compatibility for ICU's MFv1 implementation. Matching _application-defined_ enumerated data values is where it is most useful today and it is a powerful (and often necessary) tool for developers to provide message selection in place of concatenation.

I think it is super unfortunate that gender selection is often the example given for a select format. Gender and other grammatical variation stuff really needs to depend on something standardized and better defined, much like the way that `plural` was created.

Reply:
Done: #425

Reply:
My "rules of thumb" to include something or not were:
1. is already supported by Fluent & MessageFormat
2. the "building blocks" supported by ECMAScript & ICU (think formatters, PluralRules)
3. known to be useful (might be a duplicate of 1. If both Fluent & MFv1 have it, it's probably useful)
4. relatively easy to implement

That's what I had in mind when I wrote this.

---

> I think we agree on the ultimate behaviour we want to enable, but disagree about some of the details in how to achieve it.

I am not aware of such a disagreement on plain text select, which is basic string compare.